### ISSUE TITLE
Improper transparency sorting on some foliage

### ISSUE THREAD

**Issue by Ophiolith** [zesterer/openmw-volumetric-clouds#4]
I'm pretty sure this could actually be a limitation of OpenMW's postprocessing implementation, and frankly I'm not even sure if it's fixable, but nevertheless thought it worth drawing attention to - this is probably less noticeable with custom foliage, but at least on the default set there are notable transparency artifacts:
![transparency](https://user-images.githubusercontent.com/24281932/170882507-7b676f92-a955-4412-b5c6-7450a1c231fe.PNG)

![transparency3](https://user-images.githubusercontent.com/24281932/170882514-1db57e94-6377-4be0-9bd8-55107222361e.PNG)

**Comment by zesterer** [comment]
Unfortunately, there's not a huge amount I can do about this. The cloud shader uses the value in the depth buffer at each pixel to determine the distance light travels through the air, but it can't differentiate between an opaque surface and a translucent surface that has been blended with the background, because this information no longer exist. I'll keep the issue open because it would be nice to at least somewhat mitigate this somehow.

**Comment by glassmancody** [comment]
Yeah, blended trees (and other transparents) are re-drawn with alpha testing enabled to populate depth buffer.

If we didn't do this the entire leaf structure would be sky.

**Comment by zesterer** [comment]
I think it should be possible to do it in the forward pass, but we'll need to have the transparency of the fragment be altered by the fog too. I'm also not sure whether this will have sampling-based artifacts similar to those discussed by CMAugust in the past. Something to look into, I guess. The other alternative is to use the alpha-as-cloud-blend trick I mentioned before, although it doesn't generalise well.
